File: src/main/java/io/github/chaosdave34/kitpvp/cosmetics/impl/SimpleProjectileTrail.java
package io.github.chaosdave34.kitpvp.cosmetics.impl;

import io.github.chaosdave34.kitpvp.cosmetics.ProjectileTrail;
import org.bukkit.Location;
import org.bukkit.Material;
import org.bukkit.Particle;

public class SimpleProjectileTrail extends ProjectileTrail {
    protected Particle particle;

    public SimpleProjectileTrail(String id, String name, Particle particle, int levelRequirement, Material icon) {
        super(id, name, levelRequirement, icon);
        this.particle = particle;
    }

    @Override
    public void playEffect(Location location) {
        location.getWorld().spawnParticle(particle, location, 1, 0, 0, 0, 0);
    }
}

File: src/main/kotlin/io/github/chaosdave34/kitpvp/cosmetics/impl/SimpleProjectileTrail.kt
package io.github.chaosdave34.kitpvp.cosmetics.impl

import io.github.chaosdave34.kitpvp.cosmetics.ProjectileTrail
import org.bukkit.Location
import org.bukkit.Material
import org.bukkit.Particle

class SimpleProjectileTrail(id: String, name: String, var particle: Particle, levelRequirement: Int, icon: Material) :
    ProjectileTrail(id, name, levelRequirement, icon) {
    override fun playEffect(location: Location) {
        location.world.spawnParticle(particle, location, 1, 0.0, 0.0, 0.0, 0.0)
    }
}